src/content/en/2022/pwa.md

```diff
@@ -1,22 +1,517 @@
+
+The A2HS scenario has evolved into web apps that can be fully installed and deeply integrated into the host OS, in both mobile and desktop contexts. These past 12 months have seen browsers take important steps towards making sure that PWAs have a tight integration with desktop platforms, and many of the new additions to this year’s almanac reflect these changes. This is the state of PWAs in 2022.
+
+Note: As a set of web technologies, PWAs are not isolated from the rest of the web platform. While there is a chapter dedicated to Capabilities, this year we have investigated the intersection of some of these new advanced capabilities when used inside a PWA.
+
+## Service Workers
+
+[Service Workers](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) are one of the 3 core technologies (alongside Manifest file and HTTPS) of PWAs and the enabler of offlining apps, getting push notifications, and doing background processing. They serve as the base for most of the advanced experiences we expect from applications. They are also being used to define data updates and for upcoming modern functionality like [widgets based on PWA technologies](https://github.com/aarongustafson/pwa-widgets#rich-widgets).
+
+While there [isn't parity between major browsers when it comes to Service Worker](https://caniuse.com/push-api) feature support, Webkit  adding support for push notifications was a huge milestone. Earlier this year it was announced that [Apple had made changes](https://webkit.org/blog/12945/meet-web-push/) to their desktop platform to support the relevant parts of the [Push API](https://developer.mozilla.org/en-US/docs/Web/API/Push_API), [Notifications API](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API) and that [Service Workers](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) would enable Web Push. They also announced the feature would be coming to their mobile platforms in 2023.
+
+### Service Worker Usage
+
+Service workers are not as common as other core technologies of PWAs. For comparison reasons, we have run the same queries as last year, which allows us to try to make sense of the evolution of service worker usage. You can read the explanation of [why it isn’t trivial to find out actual usage of service worker](https://almanac.httparchive.org/en/2021/pwa#service-workers-usage),
+
+* Lighthouse detects a 1.6% (mobile) and 1.7% (desktop) of all websites employ a service worker. We expect this is lower than the real-world percentage due to [additional checks](https://web.dev/service-worker) that Lighthouse takes into consideration.
+* Following the same [metrics introduced last year](https://github.com/HTTPArchive/legacy.httparchive.org/blob/master/custom_metrics/pwa.js), usage of a Service Worker in websites comes up to 1.63% on desktop and 1.81% on mobile.
+
+{{ figure_markup(
+  image="sw_controlled_pages_rank.png",
```

Comment:
These images should be auto-generated using the [Puppeteer script](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/src/README.md#generating-chapter-images) and added to the PR